src/hooks/useSigningKeys.ts

```diff
@@ -0,0 +1,87 @@
+import { ethers } from "ethers";
+import { useState, useEffect, useContext } from "react";
+import { ErrorContext } from "common/context/ErrorContext";
+
+import { recoverPublicKey } from "features/vote/helpers/recoverPublicKey";
+import { derivePrivateKey } from "features/vote/helpers/derivePrivateKey";
+import usePrevious from "common/hooks/usePrevious";
+
+// User has to sign once per round.
+// public and private signing keys is indexed by address, and we ask them to resign if the roundMessage changes.
+export interface SigningKeys {
+  [address: string]: {
+    publicKey: string;
+    privateKey: string;
+    roundMessage: string;
+  };
+}
+
+export default function useSigningKeys(
+  signer: ethers.Signer | null,
+  address: string | null,
+  roundId: string
+) {
+  const storageKeys = localStorage.getItem("signingKeys");
```

Comment:
reading from localstorage is possibly expensive, we could batch this call with the `useState` below and use the initialization function of `useState`, which would only run on mount, for ex
```
const [keys, setKeys] = useState(() => {
const storage = localStorage.getItem("keys");
// ... other stuff
})

Reply:
Sounds good, done!